File: app/src/main/java/com/abproject/tsp_cart/TSPApplication.java
package com.abproject.tsp_cart;

import android.app.Application;

import dagger.hilt.android.HiltAndroidApp;
import timber.log.Timber;

@HiltAndroidApp
public class TSPApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        //initialize Timber for Project.
        Timber.plant(new Timber.DebugTree());
    }
}

File: app/src/main/java/com/abproject/TSPApplication.kt
package com.abproject

import android.app.Application
import com.abproject.tsp_cart.model.repository.BaseRepository
import com.abproject.tsp_cart.model.repository.BaseRepositoryImpl
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber
import javax.inject.Inject

@HiltAndroidApp
class TSPApplication : Application() {

    @Inject
    lateinit var baseRepository: BaseRepository

    override fun onCreate() {
        super.onCreate()

        Timber.plant(Timber.DebugTree())

        loadApplicationData()
    }


    private fun loadApplicationData() {
        baseRepository.loadUserDataFromSharedPrefs()
    }
}